app/uk/gov/hmrc/selfservicetimetopay/controllers/calculator/AmountsDueController.scala

```diff
@@ -17,34 +17,40 @@
 package uk.gov.hmrc.selfservicetimetopay.controllers.calculator
 
 import play.api.mvc._
-import uk.gov.hmrc.play.frontend.controller.FrontendController
-import uk.gov.hmrc.selfservicetimetopay.connectors.SessionCacheConnector
+import uk.gov.hmrc.selfservicetimetopay.config.TimeToPayController
 import uk.gov.hmrc.selfservicetimetopay.forms.CalculatorForm
-import uk.gov.hmrc.selfservicetimetopay.models.CalculatorAmountsDue
-import views.html.selfservicetimetopay.calculator.amounts_due_form
+import uk.gov.hmrc.selfservicetimetopay.models.{CalculatorAmountsDue, TTPSubmission}
+import uk.gov.hmrc.selfservicetimetopay.modelsFormat._
+import views.html.selfservicetimetopay.calculator._
 
 import scala.concurrent.Future
 
-class AmountsDueController(sessionCache: SessionCacheConnector) extends FrontendController {
+class AmountsDueController extends TimeToPayController {
```

Comment:
Example use of the sessionCache.

The template doesn't make use of the data, but using debugging it's present and builds up as entries are added.